### PR TITLE
[Custom Attribute] - Fixed when is empty value

### PR DESCRIPTION
**Pull request by eduardogomez97**
The problem is that the min-hago-test data is removed because the array_filter function is configured to filter out values that are considered “empty” (including 0, which in PHP evaluates to empty).

Since the value of min-hago-test is 0, the array_filter function removes it.
To avoid this, we can modify the array_filter function so that it only filters out null or empty values, but not those that are 0.

Before:
```
[
    {
        "average_rating": "0.00",
        "catalog_visibility": "visible",
        "categories": [
            "Sin categorizar"
        ],
        "id": 32,
        "price": 20,
        "purchasable": true,
        "regular_price": 20,
        "status": "publish",
        "slug": "product-test-2",
        "stock_status": "instock",
        "df_indexable": "auto",
        "image_link": "https://9314c37b-0ef5-499b-b938-d87b1c691106.clouding.host/wp-content/uploads/woocommerce-placeholder-300x300.png",
        "availability": "in stock",
        "title": "Product test 2 update on save",
        "link": "https://9314c37b-0ef5-499b-b938-d87b1c691106.clouding.host/producto/product-test-2/"
    }
]
```

After:

```
[
    {
        "average_rating": "0.00",
        "catalog_visibility": "visible",
        "categories": [
            "Sin categorizar"
        ],
        "description": "",
        "id": 32,
        "price": 20,
        "purchasable": true,
        "regular_price": 20,
        "sale_price": "",
        "short_description": "",
        "sku": "",
        "status": "publish",
        "slug": "product-test-2",
        "tags": [],
        "stock_status": "instock",
        "df_indexable": "auto",
        "min-hago-test": "0",
        "image_link": "https://9314c37b-0ef5-499b-b938-d87b1c691106.clouding.host/wp-content/uploads/woocommerce-placeholder-300x300.png",
        "availability": "in stock",
        "title": "Product test 2 update on save",
        "link": "https://9314c37b-0ef5-499b-b938-d87b1c691106.clouding.host/producto/product-test-2/"
    }
]
```

### CODE REVIEW

File: doofinder-for-woocommerce/includes/api/endpoints/class-endpoint-product.php [hunk 1]
```diff
@@ -578,7 +578,7 @@ private static function clean_fields( $product ) {
 		$product = array_filter(
 			$product,
 			function ( $value ) {
-				return ! empty( $value );
+				return $value !== null;
```
Comment (davidmolinacano) [inline doofinder-for-woocommerce/includes/api/endpoints/class-endpoint-product.php]:
```suggestion
				return ! is_null( $value );
```

File: doofinder-for-woocommerce/includes/api/endpoints/class-endpoint-product.php [hunk 2]
```diff
@@ -578,7 +578,7 @@ private static function clean_fields( $product ) {
 		$product = array_filter(
 			$product,
 			function ( $value ) {
-				return ! empty( $value );
+				return null !== $value;
```
Comment (davidmolinacano) [inline doofinder-for-woocommerce/includes/api/endpoints/class-endpoint-product.php:581]: It's also valid as:

```suggestion
				return ! is_null( $value );
```

**Review by davidmolinacano — APPROVED** [review]
LGTM!! :trophy: :medal_military:

**Review by sofia-doofinder — APPROVED** [review]

**Review by pedromcp90 — COMMENTED** [review]
Good Job!!

**Review by pedromcp90 — APPROVED** [review]
Good Job!!

**Pull request merged**